### PR TITLE
BugFix: Checks to populate matrix input to Newman's app were reversed

### PR DESCRIPTION
**Pull request by Lucas-Carmo**
This PR is ready to be merged.

**Feature or improvement description**

Only half of the direction pairs of the WAMIT mean drift files are needed to use Newman's approximation, as the other pairs can be obtained by symmetry. However, the checks used to exploit this symmetry and populate the whole matrix were reversed, and this is fixed by this pull request.

**Related issue, if one exists**

#1096 

**Impacted areas of the software**

HydroDyn (only when Newman's approximation is used)

**Additional supporting information**

**Test results, if applicable**

I may be mistaken, but there are currently no tests using Newman's approximation.

### CODE REVIEW

File: modules/hydrodyn/src/WAMIT2.f90 [hunk 1]
```diff
@@ -4221,12 +4221,12 @@ SUBROUTINE Read_DataFile3D( InitInp, Filename3D, Data3D, ErrStat, Errmsg )
                         IF ( EqualRealNos( Data3D%WvDir1(J), Data3D%WvDir2(J) ) .AND. &
                              EqualRealNos( Data3D%WvDir1(K), Data3D%WvDir2(K) ) ) THEN
 
-                              ! Check if not filled
-                           IF ( .NOT. Data3D%DataMask( I, J, K, L )  ) THEN
+                              ! Check if filled
+                              IF ( Data3D%DataMask( I, J, K, L )  ) THEN
```
Comment (andrew-platt) [inline modules/hydrodyn/src/WAMIT2.f90:4225]: Could you adjust the whitespace on this line?  The `IF` can move over 3 places left

**Comment by Lucas-Carmo** [comment]
My pleasure!

File: modules/hydrodyn/src/WAMIT2.f90 (hunk 1 above)
Comment (Lucas-Carmo) [reply modules/hydrodyn/src/WAMIT2.f90:4225]: Did it work?

Comment (andrew-platt) [reply modules/hydrodyn/src/WAMIT2.f90:4225]: My apologies, I wasn't specific enough on this.  Line 4225 needed 3 spaces deleted from before the `IF`.  The other lines are good.

Comment (Lucas-Carmo) [reply modules/hydrodyn/src/WAMIT2.f90:4225]: Sorry, I edited from my phone and did not see the rest of the file. I am gonna fix it properly when I get home.

Comment (Lucas-Carmo) [reply modules/hydrodyn/src/WAMIT2.f90]: I think it is fixed now!

**Pull request merged**